### DISCUSSION TITLE
page.images  leads to TypeError: argument of type 'NullObject' is not iterable

### DISCUSSION THREAD

**Discussion by commuter77** [py-pdf/pypdf#1732] [General]
I tried the sample code from
https://pypdf2.readthedocs.io/en/stable/user/extract-images.html
but I got TypeError: argument of type 'NullObject' is not iterable.
Anyone has the same issue ?
I confirmed I've an image in my pdf file.

the error in details:

```
testpdf2.py", line 14, in <module>
    num_of_images = len(page.images())
  File "C:\mySoft\Python39\lib\site-packages\PyPDF2\_page.py", line 481, in images
    extension, byte_stream = _xobj_to_image(x_object[obj])
  File "C:\mySoft\Python39\lib\site-packages\PyPDF2\filters.py", line 576, in _xobj_to_image
    data = x_object_obj.get_data()  # type: ignore
  File "C:\mySoft\Python39\lib\site-packages\PyPDF2\generic\_data_structures.py", line 827, in get_data
    decoded._data = decode_stream_data(self)
  File "C:\mySoft\Python39\lib\site-packages\PyPDF2\filters.py", line 538, in decode_stream_data
    data = CCITTFaxDecode.decode(data, stream.get(SA.DECODE_PARMS), height)
  File "C:\mySoft\Python39\lib\site-packages\PyPDF2\filters.py", line 463, in decode
    parms = CCITTFaxDecode._get_parameters(decode_parms, height)
  File "C:\mySoft\Python39\lib\site-packages\PyPDF2\filters.py", line 441, in _get_parameters
    if CCITT.COLUMNS in decode_parm:
TypeError: argument of type 'NullObject' is not iterable
```

**Comment by pubpub-zz** [comment]
this could be a robustness issue, can you share the failing PDF ?

**Reply by commuter77** [reply]
enclosed the PDF file.
https://1drv.ms/b/s!Ap8hS8QhqsvljWlZ4OXi8nVeVLLi?e=qbK1j3

**Reply by MartinThoma** [reply]
@commuter77 Thank you for sharing this PDF. Would it be OK if we added it to our test suite? If that is the case, please updload it to https://github.com/py-pdf/pypdf/issues/1737 :-)

**Comment by pubpub-zz** [comment]
Thanks for sharing. I confirm that the problem is linked with your pdf file and this is a bug : multiple filters is not processed correctly. I'm finalizing the fix. May I use your file to complete the test suite ?
I will close this discussion and convert it into a issue.
